### PR TITLE
Move Antiforgery implementation details to `.Internal`

### PR DESCRIPTION
**Pull request by dougbu**
- move the files
- update the namespaces

### CODE REVIEW

**Comment by dougbu** [comment]
/cc @rynowak
FYI @Eilon

**Comment by dougbu** [comment]
Recommend looking at the namespace changes and checking out what's left in
- https://github.com/aspnet/Antiforgery/tree/dougbu/be.internal/src/Microsoft.AspNetCore.Antiforgery
- https://github.com/aspnet/Antiforgery/tree/dougbu/be.internal/test/Microsoft.AspNetCore.Antiforgery.Test

File: src/Microsoft.AspNetCore.Antiforgery/Internal/IAntiforgeryAdditionalDataProvider.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 using Microsoft.AspNetCore.Http;
 
-namespace Microsoft.AspNetCore.Antiforgery
+namespace Microsoft.AspNetCore.Antiforgery.Internal
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Antiforgery/Internal/IAntiforgeryAdditionalDataProvider.cs]: Do we want to leave this one public? This is the one documented extensibility point that I'm aware of: http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages#_Scenarios_with_special

**Comment by rynowak** [comment]
:shipit: after resolving what to do with the additonal data provider

File: src/Microsoft.AspNetCore.Antiforgery/Internal/IAntiforgeryAdditionalDataProvider.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Antiforgery/Internal/IAntiforgeryAdditionalDataProvider.cs]: :ok: forgot about that case. Will move it back to top-level namespace.

**Comment by dougbu** [comment]
705c080

**Pull request closed without merging**